modules/tinymce/src/plugins/paste/main/ts/core/SmartPaste.ts

```diff
@@ -31,8 +36,10 @@ const isAbsoluteUrl = function (url: string) {
   return /^https?:\/\/[\w\?\-\/+=.&%@~#]+$/i.test(url);
 };
 
-const isImageUrl = function (url: string) {
-  return isAbsoluteUrl(url) && /.(gif|jpe?g|png)$/.test(url);
+const isImageUrl = function (editor: Editor, url: string) {
+  const escapedFileTypes = Arr.map(Settings.allowedImageFileTypes(editor), escapeRegExp).join('|');
```

Comment:
This is a purely hunch-based performance comment, so feel free to ignore, but I'm not a fan of regexes for this purpose — especially when they're regexes that are being compiled based on dynamic strings rather than nice-and-easy-to-cache static regexes.

Would it be easier to either
1. Just loop over the strings and check if `url.endsWith('.' + fileType)` for any of them
2. Shove all of the settings into a set and then check if that set contains `url.substring(1 + url.lastIndexOf('.'))` (note: this breaks if any of their file types contain a `.` so that might not be optimal if we wanted to support that)

Also, is that `.` in the old regex (and your new regex) meant to be escaped because we're looking for `.jpeg` or are we planning on accepting `'xjpeg'` as a file name

Reply:
```typescript
return isAbsoluteUrl(url) && Arr.exists(allowedFileTypes, (type) => url.endsWith('.' + type));
```